matchtools/_matchblock.py

```diff
@@ -11,91 +11,187 @@
 from geopy.distance import great_circle
 from geopy.distance import vincenty
 
+__all__ = ['MatchBlock']
 
-def check_tolerance(*dec_args):
-    """Wrapper checking whether tolerance is within the bounds."""
-    def wrapper(f):
+
+def tolerance_interval(lower=0, upper=None):
+    """Decorator checking whether tolerance is within the boundaries."""
+
+    def decorator(f):
         @wraps(f)
-        def wrapped(*args, **kwargs):
+        def wrapper(*args, **kwargs):
             if 'tolerance' in kwargs:
-                if lower_bound is not None and upper_bound is None:
-                    if kwargs['tolerance'] < lower_bound:
+                if lower is not None and upper is None:
+                    if kwargs['tolerance'] < lower:
                         msg = 'tolerance must be higher than {}'
-                        raise ValueError(msg.format(lower_bound))
-                elif lower_bound is None and upper_bound is not None:
-                    if kwargs['tolerance'] > upper_bound:
+                        raise ValueError(msg.format(lower))
+
+                elif lower is None and upper is not None:
+                    if kwargs['tolerance'] > upper:
                         msg = 'tolerance must be lower than {}'
-                        raise ValueError(msg.format(upper_bound))
-                elif lower_bound is not None and upper_bound is not None:
-                    if not (lower_bound <= kwargs['tolerance'] <= upper_bound):
+                        raise ValueError(msg.format(upper))
+
+                elif lower is not None and upper is not None:
+                    if not (lower <= kwargs['tolerance'] <= upper):
                         msg = 'tolerance must be in between {}, and {}'
-                        raise ValueError(msg.format(lower_bound, upper_bound))
+                        raise ValueError(msg.format(lower, upper))
             return f(*args, **kwargs)
-        return wrapped
+        return wrapper
+
+    if callable(lower):
+        func, lower = lower, 0
+        return decorator(func)
+    return decorator
+
 
-    if len(dec_args) == 1 and callable(dec_args[0]):
-        # no bounds provided
-        lower_bound, upper_bound = 0, None
-        return wrapper(dec_args[0])
+class Tolerance:
+    """Tolerance attributes descriptor."""
 
-    if len(dec_args) == 2 and callable(dec_args[1]):
-        # only lower bound provided
-        lower_bound, upper_bound = dec_args[0], None
-    elif len(dec_args) == 2:
-        # both bounds provided
-        lower_bound, upper_bound = dec_args[0], dec_args[1]
-    else:
-        raise ValueError('too many arguments: {}'.format(dec_args))
+    def __init__(self, attribute):
+        self.attribute = '_' + attribute
```

Comment:
I can see people use those operators in descriptors but perhaps we should consider join()?